File: apiGateway/spring-boot-microservice-3-api-gateway/src/main/java/com/sb/springbootmicroserviceapps/springbootmicroservice3apigateway/request/CourseService.java
package com.sb.springbootmicroserviceapps.springbootmicroservice3apigateway.request;

import java.util.List;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;


@FeignClient(value = "course-service", path="/api/course", //url="${course.service.url}",
			configuration = FeignConfiguration.class)
public interface CourseService 
{
	@PostMapping
	Object saveCourse(@RequestBody Object course);
	
	@DeleteMapping("/delete/{courseId}")
	void deleteCourse(@PathVariable("courseId") long courseId);
	
	@GetMapping
	List<Object> getAllCourses();

	@PatchMapping(path = "/partialUpdate/{courseId}", consumes = "application/json")
	void partialUpdateCourse(@PathVariable("courseId") long id, Object course);
}

File: frontend/react-course-seller/src/services/Course.Service.js
import axios from 'axios';
import { BASE_API_URL } from '../common/Constants';
import { authHeader } from './Base.Service';


const API_URL = BASE_API_URL + '/gateway/course';

class CourseService{

    saveCourse(course){
        return axios.post(API_URL, course, {headers: authHeader()});
    }

    deleteCourse(course){
        return axios.delete(API_URL + '/delete/' + course.id, {headers: authHeader()});
    }

    getAllCourses(){
        return axios.get(API_URL, {headers: authHeader()});
    }

    partialUpdateCourse(course){
        return axios.patch(API_URL + '/partialUpdate/' + course.id, {headers: authHeader()});
    }

}

export default new CourseService();

